From 1d1443a6fec30fd8ebc75fd7371b4ac99a4a41ca Mon Sep 17 00:00:00 2001
From: Quanyang Wang <quanyang.wang@windriver.com>
Date: Fri, 17 Jul 2020 14:49:20 +0800
Subject: [PATCH] Revert "eglinfo: Add EXT_platform_* awareness"

This reverts commit 7a0f0310a9b5 ("eglinfo: Add EXT_platform_* awareness")
and 164487d7b3a0 ("eglinfo: Learn about EGL_EXT_platform_device").

Since these 2 patches depend on the eglext.h of version 24567, but
libmali-xlnx can only provide the eglext.h of version 21116, so revert
them to fix build error.

Upstream-Status: Inappropriate

Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
---
 src/egl/opengl/eglinfo.c | 41 ++++++----------------------------------
 1 file changed, 6 insertions(+), 35 deletions(-)

diff --git a/src/egl/opengl/eglinfo.c b/src/egl/opengl/eglinfo.c
index f7da372b..0124c37c 100644
--- a/src/egl/opengl/eglinfo.c
+++ b/src/egl/opengl/eglinfo.c
@@ -115,7 +115,7 @@ PrintConfigs(EGLDisplay d)
 }
 
 
-static const char *
+static void
 PrintExtensions(EGLDisplay d)
 {
    const char *extensions, *p, *end, *next;
@@ -126,7 +126,7 @@ PrintExtensions(EGLDisplay d)
 
    extensions = eglQueryString(d, EGL_EXTENSIONS);
    if (!extensions)
-      return NULL;
+      return;
 
    column = 0;
    end = extensions + strlen(extensions);
@@ -153,8 +153,6 @@ PrintExtensions(EGLDisplay d)
 
    if (column > 0)
       printf("\n");
-
-   return extensions;
 }
 
 static int
@@ -164,7 +162,7 @@ doOneDisplay(EGLDisplay d, const char *name)
 
    printf("%s:\n", name);
    if (!eglInitialize(d, &maj, &min)) {
-      printf("eglinfo: eglInitialize failed\n\n");
+      printf("eglinfo: eglInitialize failed\n");
       return 1;
    }
 
@@ -186,43 +184,12 @@ doOneDisplay(EGLDisplay d, const char *name)
 int
 main(int argc, char *argv[])
 {
-   int ret = 0;
-   const char *clientext;
+   int ret;
 
-   clientext = PrintExtensions(EGL_NO_DISPLAY);
+   PrintExtensions(EGL_NO_DISPLAY);
    printf("\n");
 
-   if (strstr(clientext, "EGL_EXT_platform_base")) {
-       PFNEGLGETPLATFORMDISPLAYEXTPROC getPlatformDisplay =
-           (PFNEGLGETPLATFORMDISPLAYEXTPROC)
-           eglGetProcAddress("eglGetPlatformDisplayEXT");
-       if (strstr(clientext, "EGL_KHR_platform_android"))
-           ret += doOneDisplay(getPlatformDisplay(EGL_PLATFORM_ANDROID_KHR,
-                                                  EGL_DEFAULT_DISPLAY,
-                                                  NULL), "Android platform");
-       if (strstr(clientext, "EGL_MESA_platform_gbm") ||
-           strstr(clientext, "EGL_KHR_platform_gbm"))
-           ret += doOneDisplay(getPlatformDisplay(EGL_PLATFORM_GBM_MESA,
-                                                  EGL_DEFAULT_DISPLAY,
-                                                  NULL), "GBM platform");
-       if (strstr(clientext, "EGL_EXT_platform_wayland") ||
-           strstr(clientext, "EGL_KHR_platform_wayland"))
-           ret += doOneDisplay(getPlatformDisplay(EGL_PLATFORM_WAYLAND_EXT,
-                                                  EGL_DEFAULT_DISPLAY,
-                                                  NULL), "Wayland platform");
-       if (strstr(clientext, "EGL_EXT_platform_x11") ||
-           strstr(clientext, "EGL_KHR_platform_x11"))
-           ret += doOneDisplay(getPlatformDisplay(EGL_PLATFORM_X11_EXT,
-                                                  EGL_DEFAULT_DISPLAY,
-                                                  NULL), "X11 platform");
-       if (strstr(clientext, "EGL_EXT_platform_device"))
-           ret += doOneDisplay(getPlatformDisplay(EGL_PLATFORM_DEVICE_EXT,
-                                                  EGL_DEFAULT_DISPLAY,
-                                                  NULL), "Device platform");
-   }
-   else {
-      ret = doOneDisplay(eglGetDisplay(EGL_DEFAULT_DISPLAY), "Default display");
-   }
+   ret = doOneDisplay(eglGetDisplay(EGL_DEFAULT_DISPLAY), "Default display");
 
    return ret;
 }
-- 
2.17.1

